src/effects/Reverb.cpp

```diff
@@ -158,21 +158,28 @@ unsigned EffectReverb::GetAudioOutCount() const
 static size_t BLOCK = 16384;
 
 bool EffectReverb::ProcessInitialize(
-   EffectSettings &, sampleCount, ChannelNames chanMap)
+   EffectSettings& settings, sampleCount, ChannelNames chanMap)
+{
+   return InstanceInit(mMaster, chanMap, /* forceStereo= */ false);
+}
+
+
+bool EffectReverb::InstanceInit(ReverbState& state, ChannelNames chanMap, bool forceStereo)
 {
    bool isStereo = false;
-   mNumChans = 1;
-   if (chanMap && chanMap[0] != ChannelNameEOL && chanMap[1] == ChannelNameFrontRight)
+   state.mNumChans = 1;
+   if (    (chanMap && chanMap[0] != ChannelNameEOL && chanMap[1] == ChannelNameFrontRight)
+        || forceStereo )
    {
       isStereo = true;
-      mNumChans = 2;
+      state.mNumChans = 2;
    }
 
-   mP = (Reverb_priv_t *) calloc(sizeof(*mP), mNumChans);
+   state.mP = (Reverb_priv_t *) calloc(sizeof(*state.mP), state.mNumChans);
```

Comment:
Define something like this (you don't need my comments in it):
```
struct Reverb_priv_ex : Reverb_priv_t {
   Reverb_priv_ex() : Reverb_priv_t{} {} // this does zero-filling of the new structure, as calloc would
   ~Reverb_priv_ex{ reverb_delete(reverb); } };
```

Then:

```
state.mP = std::make_unique<Reverb_priv_ex[]>(state.mNumChans)
```

Make appropriate change of type of mP so this assignment works.

Reply:
Done; although it had to be `reverb_delete(&reverb)`

A consequence of using the unique_ptr in the State, was that when pushing back a slave to the slaves vector, I had to use `push_back( std::move(slave) )` instead of a simple `push_back( slave )`